pkg/apis/eksctl.io/v1alpha5/vpc.go

```diff
@@ -268,75 +268,95 @@ func DefaultCIDR() ipnet.IPNet {
 //
 // If a user specifies a subnet by AZ without CIDR and ID but multiple subnets
 // exist in this VPC, one will be arbitrarily chosen.
-func ImportSubnet(subnets AZSubnetMapping, subnet *ec2types.Subnet, makeSubnetAlias func(*ec2types.Subnet) string) error {
-	if subnets == nil {
+func ImportSubnet(subnets AZSubnetMapping, localSubnetsConfig AZSubnetMapping, subnet *ec2types.Subnet, makeSubnetAlias func(*ec2types.Subnet) string) error {
+	if localSubnetsConfig == nil {
 		return nil
 	}
 
-	subnetCIDR, err := ipnet.ParseCIDR(*subnet.CidrBlock)
+	remoteSubnet, err := remoteSubnetToAZSubnetSpec(subnet)
 	if err != nil {
-		return fmt.Errorf("unexpected error parsing subnet CIDR %q: %w", *subnet.CidrBlock, err)
-	}
-
-	var (
-		subnetID = *subnet.SubnetId
-		az       = *subnet.AvailabilityZone
-	)
-
-	subnetAlias := makeSubnetAlias(subnet)
-	if network, ok := subnets[subnetAlias]; !ok {
-		newS := AZSubnetSpec{ID: subnetID, AZ: az, CIDR: subnetCIDR}
-		// Used if we find an exact ID match
-		var idKey string
-		// Used if we match to AZ/CIDR
-		var guessKey string
-		for k, s := range subnets {
-			if s.ID == "" {
-				if s.AZ != az || (s.CIDR.String() != "" && s.CIDR.String() != subnetCIDR.String()) {
-					continue
-				}
-				if guessKey != "" {
-					return fmt.Errorf("unable to unambiguously import subnet, found both %s and %s", guessKey, k)
-				}
-				guessKey = k
-			} else if s.ID == subnetID {
-				if s.CIDR.String() != "" && s.CIDR.String() != subnetCIDR.String() {
-					return fmt.Errorf("subnet CIDR %q is not the same as %q", s.CIDR.String(), subnetCIDR.String())
-				}
-				if idKey != "" {
-					return fmt.Errorf("unable to unambiguously import subnet, found both %s and %s", idKey, k)
-				}
-				idKey = k
-			}
-		}
-		if idKey != "" {
-			subnets[idKey] = newS
-		} else if guessKey != "" {
-			subnets[guessKey] = newS
-		} else {
-			if subnet.OutpostArn != nil {
-				newS.OutpostARN = *subnet.OutpostArn
-			}
-			subnets[subnetAlias] = newS
-		}
-	} else {
-		if network.ID == "" {
-			network.ID = subnetID
-		} else if network.ID != subnetID {
-			return fmt.Errorf("subnet ID %q is not the same as %q", network.ID, subnetID)
+		return err
+	}
+
+	// if a VPC config was provided as part of the config file,
+	// we need to validate it against the remote config
+	// and return an error in case of missmatch
+	subnetKey, err := validateLocalConfigAgainstRemote(localSubnetsConfig, remoteSubnet, makeSubnetAlias(subnet))
+	if err != nil {
+		return fmt.Errorf("missmatch found between local and remote VPC config: %w", err)
```

Comment:
Minor nit here and in tests:

```suggestion
		return fmt.Errorf("mismatch found between local and remote VPC config: %w", err)
```